presto-tests/src/test/java/io/prestosql/tests/TestProcedureCreation.java

```diff
@@ -0,0 +1,197 @@
+                "methodHandle is null",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(),
+                        null));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenMethodHandleHasVarargs()
+    {
+        assertThrows(
+                IllegalArgumentException.class,
+                "Method must have fixed arity",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(),
+                        MethodHandleUtil.methodHandle(Procedures.class, "funWithVarargs", String[].class)));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenArgumentCountDoesntMatch()
+    {
+        assertThrows(
+                IllegalArgumentException.class,
+                "Method parameter count must match arguments",
+                () -> new Procedure(
+                        "schema",
+                        "name",
+                        ImmutableList.of(
+                                new Procedure.Argument("name", VARCHAR, true, null),
+                                new Procedure.Argument("name2", VARCHAR, true, null),
+                                new Procedure.Argument("name3", VARCHAR, true, null)),
+                        MethodHandleUtil.methodHandle(Procedures.class, "fun1", ConnectorSession.class, Object.class)));
+    }
+
+    private static Procedure createTestProcedure(List<Procedure.Argument> arguments)
+    {
+        int argumentsCount = arguments.size();
+        String functionName = "fun" + argumentsCount;
+
+        Class<?>[] clazzes = new Class<?>[argumentsCount + 1];
+        clazzes[0] = ConnectorSession.class;
+
+        for (int i = 0; i < argumentsCount; i++) {
+            clazzes[i + 1] = Object.class;
+        }
+
+        return new Procedure(
+                "schema",
+                "name",
+                arguments,
+                MethodHandleUtil.methodHandle(Procedures.class, functionName, clazzes));
+    }
+
+    private static <T extends Throwable> void assertThrows(Class<T> exception, String message, Runnable runnable)
+    {
+        try {
+            runnable.run();
+            fail(format("Expected thrown exception %s with message %s", exception, message));
+        }
+        catch (Throwable t) {
+            if (t instanceof AssertionError) {
+                throw t;
+            }
+
+            if (!t.getClass().equals(exception)) {
+                throw new AssertionError(format("Expected exception was %s but actual was %s", exception, t));
+            }
+
+            if (!t.getMessage().contentEquals(message)) {
+                throw new AssertionError(format("Expected exception message to be %s but was %s", message, t.getMessage()));
+            }
+        }
+    }
+
+    public static class Procedures
+    {
+        public void fun0(ConnectorSession session)
```

Comment:
Nit: format `{}` on the same line for intentionally empty methods. We do that so it's clear that the methods are intended to be empty.

Reply:
bump